templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -49,6 +47,37 @@ contents:
             show \
             "{{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP}}" \
             "{{.Infra.Status.PlatformStatus.BareMetal.IngressIP}}")
+        # check if we got DHCPv6 hostname. The value would usually be in
+        # DHCP6_FQDN_FQDN but NM internal client does not parse the option
+        # Only if it is IPv4
+        if grep -q ":" "$NAMESERVER_IP"; then
```

Comment:
Is it possible this could be an issue in dual stack? I realize we don't care about that yet, but we will. I'm wondering if it's possible we'll have ipv6 addresses where the nameserver is ipv4. Maybe we should look at IP6_NUM_ADDRESSES to detect an ipv6 env?

Reply:
When we support dual stack, I assume that there will be a PR to drop this hack and we can handle this case then :-)

Reply:
Oh, good point. We should have the NM fix by then.

Reply:
That being said, this doesn't actually do what you want. grep treats the second string as a filename, not the thing to be grepped. I think you want:

if [[ "$NAMESERVER_IP" =~ ":"]]; then

This caused my test to fail. I'm retesting with it fixed.